mantidproject/mantid
# BinaryOperation updated

**Description of work.**

- avoid use of deprecated `readX`, `readY`, `readE`
- update documentation of `WeigthedMean` (formula and its explanation similar to `Mean`)
- `WeightedMean` as `CommutativeBinaryOperation`


**To test:**

All tests should pass. Code review.

The issue is not resolved, these are intermediate changes.

*This does not require release notes* because the documentation change is minor and the code changes does not change the user algorithms.

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/IndividualTicketTesting/)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.
